comp/netflow/payload/payload.go

```diff
@@ -41,23 +43,55 @@ type ObservationPoint struct {
 
 // FlowPayload contains network devices flows
 type FlowPayload struct {
-	FlushTimestamp int64            `json:"flush_timestamp"`
-	FlowType       string           `json:"type"`
-	SamplingRate   uint64           `json:"sampling_rate"`
-	Direction      string           `json:"direction"`
-	Start          uint64           `json:"start"` // in seconds
-	End            uint64           `json:"end"`   // in seconds
-	Bytes          uint64           `json:"bytes"`
-	Packets        uint64           `json:"packets"`
-	EtherType      string           `json:"ether_type,omitempty"`
-	IPProtocol     string           `json:"ip_protocol"`
-	Device         Device           `json:"device"`
-	Exporter       Exporter         `json:"exporter"`
-	Source         Endpoint         `json:"source"`
-	Destination    Endpoint         `json:"destination"`
-	Ingress        ObservationPoint `json:"ingress"`
-	Egress         ObservationPoint `json:"egress"`
-	Host           string           `json:"host"`
-	TCPFlags       []string         `json:"tcp_flags,omitempty"`
-	NextHop        NextHop          `json:"next_hop,omitempty"`
+	FlushTimestamp   int64            `json:"flush_timestamp"`
+	FlowType         string           `json:"type"`
+	SamplingRate     uint64           `json:"sampling_rate"`
+	Direction        string           `json:"direction"`
+	Start            uint64           `json:"start"` // in seconds
+	End              uint64           `json:"end"`   // in seconds
+	Bytes            uint64           `json:"bytes"`
+	Packets          uint64           `json:"packets"`
+	EtherType        string           `json:"ether_type,omitempty"`
+	IPProtocol       string           `json:"ip_protocol"`
+	Device           Device           `json:"device"`
+	Exporter         Exporter         `json:"exporter"`
+	Source           Endpoint         `json:"source"`
+	Destination      Endpoint         `json:"destination"`
+	Ingress          ObservationPoint `json:"ingress"`
+	Egress           ObservationPoint `json:"egress"`
+	Host             string           `json:"host"`
+	TCPFlags         []string         `json:"tcp_flags,omitempty"`
+	NextHop          NextHop          `json:"next_hop,omitempty"`
+	AdditionalFields map[string]any   `json:"-"`
+}
+
+func (p FlowPayload) MarshalJSON() ([]byte, error) {
```

Comment:
This custom marshaller allows marshalling `AdditionalFields` at the root of the json payload